src/main/java/org/mockito/internal/creation/instance/DefaultInstantiatorProvider.java

```diff
@@ -12,8 +12,8 @@
     private final static Instantiator INSTANCE = new ObjenesisInstantiator();
 
     public Instantiator getInstantiator(MockCreationSettings<?> settings) {
-        if (settings != null && settings.isUsingConstructor()) {
-            return new ConstructorInstantiator(settings.getOuterClassInstance());
+        if (settings != null && settings.getConstructorArgs() != null) {
```

Comment:
A check for `settings.getConstructorArgs() != null` is not required if we check that null can't be passed to `MockSetting.useConstructor(Object ...args)` as described above.